### ISSUE TITLE
Metric computation broken when metric objects are used instead of string identifiers

### ISSUE THREAD

**Issue by razorx89** [tensorflow/neural-structured-learning#8]
Example from https://www.tensorflow.org/neural_structured_learning:
```python
import tensorflow as tf
import neural_structured_learning as nsl

# Prepare data.
(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

# Create a base model -- sequential, functional, or subclass.
model = tf.keras.Sequential([
    tf.keras.Input((28, 28), name='feature'),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(128, activation=tf.nn.relu),
    tf.keras.layers.Dense(10, activation=tf.nn.softmax)
])

# Wrap the model with adversarial regularization.
adv_config = nsl.configs.make_adv_reg_config(multiplier=0.2, adv_step_size=0.05)
adv_model = nsl.keras.AdversarialRegularization(model, adv_config=adv_config)

# Compile, train, and evaluate.
adv_model.compile(optimizer='adam',
                  loss='sparse_categorical_crossentropy',
                  metrics=[tf.keras.metrics.SparseCategoricalAccuracy()])
adv_model.fit({'feature': x_train, 'label': y_train}, batch_size=32, epochs=5)
adv_model.evaluate({'feature': x_test, 'label': y_test})
```

Instead of passing 'accuracy' (string) as metric, an instance of the underlying metric class is passed to the compile function. It seems like the metric never gets updated, since the accuracy value stays at 0.0 while at the same time the loss decreases. This also happens with metrics which are not part of the default keras metric set (e.g. from the tensorflow_addons repository).

Package versions:
tensorflow-gpu                2.0.0rc0
neural-structured-learning    1.0.0

**Comment by csferng** [comment]
@razorx89 Thanks for reporting. I am looking into this.

**Comment by csferng** [comment]
@razorx89 The problem should be fixed now at the master branch, and the fix will be included in the next release. Please let us know if you see any other problem. Thanks.
